src/main/kotlin/com/spectralogic/rioBench/commands/Benchmark.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * **************************************************************************
+ * Copyright 2014-2019 Spectra Logic Corporation. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ * this file except in compliance with the License. A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file.
+ * This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ * **************************************************************************
+ */
+
+package com.spectralogic.rioBench.commands
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.required
+import com.github.ajalt.clikt.parameters.types.choice
+import com.github.ajalt.clikt.parameters.types.long
+import com.spectralogic.escapepod.devintegration.FileToArchive
+import com.spectralogic.escapepod.devintegration.FilesToArchive
+import com.spectralogic.escapepod.devintegration.RioClient
+import com.spectralogic.escapepod.devintegration.UserLoginCredentials
+import com.spectralogic.escapepod.devintegration.createRioAuthClient
+import com.spectralogic.escapepod.devintegration.createRioClient
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.joinAll
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import java.net.URI
+import java.time.Instant
+import kotlin.math.pow
+import kotlin.math.roundToLong
+
+class Benchmark : CliktCommand(help = "Run the benchmark suite", name = "benchmark") {
+    private val host: String by option("-h", "--hostname", help = "Rio hostname").required()
+    private val broker: String by option("-b", "--broker", help = "Name of the broker to use").required()
+    private val jobs: Long by option("-j", "--jobs", help = "number of jobs to launch").long().required()
+    private val number: Long by option("-n", "--number", help = "number of items per job").long().required()
+    private val size: Long by option("-s", "--size", help = "size of an individual item").long().required()
+    private val units: Int by option("-u", "--units", help = "units for job item")
+        .choice(Pair("b", 0), Pair("kb", 1), Pair("Mb", 2), Pair("Gb", 3), Pair("Tb", 4))
+        .required()
+
+    override fun run() {
+        runBlocking {
+            val timedRioClient = TimedRioClient(host, UserLoginCredentials("spectra", "spectra"))
+            val itemSize: Long = size * (1024.0.pow(units)).roundToLong()
+            (0 until jobs).map { job ->
```

Comment:
Does this create a sequence or a list?  And I believe there is an alternative way to do this build into kotlin based around ranges or something.  I'll need to look and get back to you.

Reply:
Oh good point, until returns a range not a sequence

Reply:
I think this should be better, but I still have the two-list I'd like to figure out a way to make this better.